src/components/MealCostCalculator.vue

```diff
@@ -0,0 +1,304 @@
+            <td>{{portionGrossCost()}}</td>
+          </tr>
+        </tbody>
+      </table>
+    <button class="dim button">SAVE</button>
+  </div>
+</template>
+
+<script>
+import { uuid } from 'vue-uuid'; 
+export default {
+  name: "MealCostCalculator",
+   data() {
+    return {
+      meal: {
+        name:"Bigos",
+        id:1,
+        ingredients:[
+          {
+          id:1,
+          name:"Lopatka",
+          quantity:4.1,
+          price:10.45,
+          vat:"1", 
+          gross:0,
+          net:0,
+          },
+          {
+          id:2,
+          name:"Kielbasa",
+          quantity:1.6,
+          price:20.99,
+          vat:"1",
+          gross:0,
+          net:0,
+          },
+        ],
+        portions: {
+          totalWeight: "",
+          portionWeight: "",
+          portionsCount:"",
+          portionNetCost: 0,
+          portionGrossCost: 0,
+        },
+        totalNetCost: 0,
+        totalGrossCost: 0,
+        totalPortionNetCost: 0,
+        totalportionGrossCost: 0,
+      },
+
+      vatOptions: [
+        { text: "Select Vat", value:1, disabled: true},
+        { text: "5", value:1.05},
+        { text: "8", value:1.08},
+        { text: "23", value:1.23}      
+      ],
+      value: "",
+      tHeader: [
+        {id: 1, name: "Ingredient"}, 
+        {id: 2, name: "Price"}, 
+        {id: 3, name: "Weight/kg"}, 
+        {id: 4, name: "VAT"}, 
+        {id: 5, name: "Net value"}, 
+        {id: 6, name: "Gross value"} 
+      ],
+    }
+  },
+  
+  methods: {
+      calculateNetValue(quantity, price, ingredientIndex) {
+        const netValue =  (quantity * price).toFixed(2);
+        this.meal.ingredients[ingredientIndex].net = netValue
+        return netValue;
+      },
+
+      calculateGrossValue(quantity,price,vat, ingredientIndex) {
+        const grossValue = (quantity * price * vat).toFixed(2);
+        this.meal.ingredients[ingredientIndex].gross = grossValue;
+
+        return grossValue;
+      },
+
+      handleInput(event) {
+        this.value = event.target.value;
+      },
+
+      addIngredient() {
+        const newID = uuid.v4();
+
+        const newIngredient = {
+          id: newID,
+          name: this.value,
+          quantity: "",
+          price: "",
+          vat: "1",
+          gross: "",
+          net: "",
+        };
+
+        if(this.value) {
+          this.meal.ingredients.push(newIngredient);
+          this.value = "";
+        }
+      },
+
+      calculatePortionsCount(totalWeight, portionWeight) {
+        let portionCount = 0;
+
+        if ((this.meal.portions.totalWeight == "" && this.meal.portions.totalWeight == 0) || (this.meal.portions.portionWeight == "" && this.meal.portions.portionWeight == 0)) {
+          portionCount = 0;
+        } 
+        else {
+          portionCount = Math.round(totalWeight / (portionWeight / 1000));
+          this.meal.portions.portionsCount = portionCount;
+        }
```

Comment:
this would always evaluate to `false`:
`this.meal.portions.totalWeight == "" && this.meal.portions.totalWeight == 0`

```
if(this.meal.portion.totalWeight || this.meal.portions.portionWeight) {
   portionCount = Math.round(totalWeight / (portionWeight / 1000));
   this.meal.portions.portionsCount = portionCount;
}
```